src/vma/sock/sockinfo_udp.cpp

```diff
@@ -1973,12 +1972,168 @@ bool sockinfo_udp::rx_input_cb(mem_buf_desc_t* p_desc, void* pv_fd_ready_array)
+	}
+
+	// convert hw timestamp to system time
+	if (m_n_tsing_flags & SOF_TIMESTAMPING_RAW_HARDWARE) {
+		ring_simple* owner_ring = (ring_simple*) p_desc->p_desc_owner;
+		if (owner_ring) {
+			owner_ring->convert_hw_time_to_system_time(p_desc->rx.hw_raw_timestamp, &p_desc->rx.udp.hw_timestamp);
+		}
+	}
+
+	vma_recv_callback_retval_t cb_ret = VMA_PACKET_RECV;
+	if (m_rx_callback && ((cb_ret=inspect_by_user_cb(p_desc))==VMA_PACKET_DROP)) {
+		si_udp_logfunc("rx packet discarded - by user callback");
+		return false;
+	}
+	// Yes, we want to keep this packet!
+	// And we must increment ref_counter before pushing this packet into the ready queue
+	//  to prevent race condition with the 'if( (--ref_count) <= 0)' in ib_comm_mgr
+	p_desc->inc_ref_count();
+
+#ifdef DEFINED_VMAPOLL
+	if (p_desc->rx.vma_polled) {
+		fill_completion(p_desc);
+		p_desc->rx.vma_polled = false;
+	} else
+#endif
+	{
+		update_ready(p_desc, pv_fd_ready_array, cb_ret);
+	}
+	return true;
+}
+
+/**
+ *	Performs inspection and processing for simple UC UDP case
+ *	bypassing all other inspections
+ */
+inline bool sockinfo_udp::rx_process_udp_uc_packet(mem_buf_desc_t* p_desc, void* pv_fd_ready_array)
+{
+	if (!inspect_uc_packet(p_desc))
+		return false;
+
+	// keep the sw_timestamp the same to all sockets
+	if ((m_b_rcvtstamp ||
+		 (m_n_tsing_flags &
+		  (SOF_TIMESTAMPING_RX_SOFTWARE | SOF_TIMESTAMPING_SOFTWARE))) &&
+		!p_desc->rx.udp.sw_timestamp.tv_sec) {
+		clock_gettime(CLOCK_REALTIME, &(p_desc->rx.udp.sw_timestamp));
+	}
+
+	// convert hw timestamp to system time
+	if (m_n_tsing_flags & SOF_TIMESTAMPING_RAW_HARDWARE) {
+		ring_simple* owner_ring = (ring_simple*) p_desc->p_desc_owner;
+		if (owner_ring) {
+			owner_ring->convert_hw_time_to_system_time(p_desc->rx.hw_raw_timestamp, &p_desc->rx.udp.hw_timestamp);
+		}
+	}
 
-return true;
+	vma_recv_callback_retval_t cb_ret = VMA_PACKET_RECV;
+	if (m_rx_callback && ((cb_ret=inspect_by_user_cb(p_desc))==VMA_PACKET_DROP)) {
+		si_udp_logfunc("rx packet discarded - by user callback");
+		return false;
+	}
+	// Yes, we want to keep this packet!
+	// And we must increment ref_counter before pushing this packet into the ready queue
+	//  to prevent race condition with the 'if( (--ref_count) <= 0)' in ib_comm_mgr
+	p_desc->inc_ref_count();
+
+#ifdef DEFINED_VMAPOLL
+	if (p_desc->rx.vma_polled) {
+		fill_completion(p_desc);
+		p_desc->rx.vma_polled = false;
+	} else
+#endif
+	{
+		update_ready(p_desc, pv_fd_ready_array, cb_ret);
+	}
+	return true;
+}
+
+/**
+ *	set packet inspector and processor
+ */
+inline void sockinfo_udp::set_rx_packet_processor(void)
+{
+	si_udp_logdbg("was_connected: %d mapped: %d multicast: %d",
+		      m_was_connected, m_mapped, m_multicast);
+	// fallback to full inspection in case any of stream modes were set
+	// m_was_connected for inspection of SRC port/addr (connect for UDP)
+	// m_mapped - set by specific to VMA sockopt
+	// and m_multicast stream after joining to MC group.
```

Comment:
I propose the following message:

```
	// Select partial or full packet processing.
	// Full packet processing is selected in case of:
	// - connect() was done on the UDP socket
	//   In this case the UDP 3-tuple is not sufficient for the packet maching.
	// - In the case that socket mapping is enabled extra processing is required
	// - Multicast packets
	// For simple UC traffic reduced packet processing is selected.

```